recipes/tcl_installer/8.6.10/conanfile.py

```diff
@@ -0,0 +1,202 @@
+        win_config_dir = self._get_configure_dir("win")
+        win_makefile_vc = os.path.join(win_config_dir, "makefile.vc")
+        tools.replace_in_file(win_makefile_vc, "@type << >$@", "type <<temp.tmp >$@")
+
+
+    def config_options(self):
+        if self.settings.os_build == "Windows":
+            del self.options.fPIC
+
+    def _get_default_build_system(self):
+        if self.settings.os_build == "Macos":
+            return "macosx"
+        elif self.settings.os_build == "Linux":
+            return "unix"
+        elif self.settings.os_build == "Windows":
+            return "win"
+        else:
+            raise ConanInvalidConfiguration("Unknown settings.os_build={}".format(self.settings.os))
+
+    def _get_configure_dir(self, build_system=None):
+        if build_system is None:
+            build_system = self._get_default_build_system()
+        if build_system not in ["win", "unix", "macosx"]:
+            raise ConanInvalidConfiguration("Invalid build system: {}".format(build_system))
+        return os.path.join(self.source_folder, self._source_subfolder, build_system)
+
+    def _get_auto_tools(self):
+        autoTools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        return autoTools
+
+    def _build_nmake(self, targets):
+        opts = []
+        # https://core.tcl.tk/tips/doc/trunk/tip/477.md
+        if not self.options.shared:
+            opts.append("static")
+        if self.settings.build_type == "Debug":
+            opts.append("symbols")
+        if "MD" in self.settings.compiler.runtime:
+            opts.append("msvcrt")
+        else:
+            opts.append("nomsvcrt")
+        if "d" not in self.settings.compiler.runtime:
+            opts.append("unchecked")
+        vcvars_command = tools.vcvars_command(self.settings)
+        self.run(
+            '{vcvars} && nmake -nologo -f "{cfgdir}/makefile.vc" INSTALLDIR="{pkgdir}" OPTS={opts} {targets}'.format(
+                vcvars=vcvars_command,
+                cfgdir=self._get_configure_dir("win"),
+                pkgdir=self.package_folder,
+                opts=",".join(opts),
+                targets=" ".join(targets),
+            ), cwd=self._get_configure_dir("win"),
+        )
+
+    def _build_autotools(self):
+        conf_args = [
+            "--enable-threads",
+            "--enable-shared" if self.options.shared else "--disable-shared",
+            "--enable-symbols" if self.settings.build_type == "Debug" else "--disable-symbols",
+            "--enable-64bit" if self.settings.arch_build == "x86_64" else "--disable-64bit",
+            "--prefix={}".format(self.package_folder),
+        ]
+        autoTools = self._get_auto_tools()
+        autoTools.configure(configure_dir=self._get_configure_dir(), args=conf_args, vars={"PKG_CFG_ARGS": " ".join(conf_args)})
+
+        # https://core.tcl.tk/tcl/tktview/840660e5a1
+        for root, _, files in os.walk(self.build_folder):
```

Comment:
I haven't checked, but I think this logic can be moved to `_fix_sources` and by modifying `Makefile.in` instead of `Makefile`.  But I'm not entirely sure!

Reply:
Nevermind, this is too hard because paths are also present in definitions and in subpaths.
This gets ugly fast.